### PR TITLE
Initial Storybook Docs Setup with EzChip

### PR DESCRIPTION
**Pull request by noranda**
_Note: This is the first of many planned PRs to be merged into the feature branch `storybook-docs`._

_Note: This removes `EzChip` from the current Gatsby doc site. Before merging the feature branch `storybook-docs` into `main`, the current Gatsby site should be decommissioned and the new Storybook Docs site deployed._

## What did we change?

This PR implements the initial Storybook Docs setup to eventually move over our current Gatsby [doc site](https://ezcater.github.io/recipe/) to [Storybook Docs](https://storybook.js.org/docs/react/writing-docs/introduction).
- Implements initial Storybook Docs setup with `EzChip`
    - Adds/replaces `EzChip` documentation, stories, snapshot, and regression tests
- Adds and configures the following storybook docs addons: actions, controls, measure, outline, interactions, theming, a11y, links, and playroom
- Adds a Welcome page with `Intro` components and all current guides
- Restructures the Theming guide to use Storybook Docs `ColorPalette` and `Typeset` doc blocks
- Adds a `TableOfContents` component for use in doc pages (TODO: add `scrollspy`)
- Adds a `ComponentLibrary` component for showcasing all our components
- Changes how we do snapshots - we will no longer need to use the `snapshots` package - simply render the component in `[ComponentName]Snapshot.tsx` and the `ComponentLibrary` component will automatically find and render it in the component library
- Adds a `STORYBOOK_INCLUDE_REGRESSION_STORIES` environment variable to handle showing the regression stories in our Chromatic tests, but not in production
  - To run in development: `STORYBOOK_INCLUDE_REGRESSION_STORIES=1 yarn storybook`
  
TODO:
- Convert all remaining components to storybook docs
- Fix `TableOfContents` component to use `scrollspy`
- Remove Gatsby doc site and deploy new Storybook Docs site

## Why are we doing this?

Our current doc site, built in Gatsby, is difficult to maintain and scale. We also want our doc site to help support our “recipe first” initiative by allowing our designers & devs interact more easily with our components. Storybook Docs solves these issues for us. It is the industry standard for design systems with features like:
      - Automatically embed Storybook stories
      - Basic documentation built in with options to customize
      - Large library of useful addons like controls for props, interaction and accessibility testing, and playroom
      - Built in, comprehensive search
      - MDX support integrated with Storybook’s UI and tooling

## Screenshot(s) / Gif(s):
Screen Shot 2022-10-28 at 1 24 51 PM
Screen Shot 2022-10-28 at 1 25 05 PM
Screen Shot 2022-10-28 at 1 25 28 PM
Screen Shot 2022-10-28 at 1 25 54 PM
Screen Shot 2022-10-28 at 1 26 09 PM

## Checklist

- [x] Provide test coverage for the changes made
- [x] Create a changeset (`yarn changeset`) with a summary of the changes

### CODE REVIEW

File: packages/recipe/src/components/EzIcon/EzIcon.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ const EzIcon = forwardRef<Ref, EzIconProps>(({color, icon, size, ...props}, ref)
   const theme = useTheme();
   const iconProps = {
     color,
-    fontSize: size === 'inherit' ? size : theme.typography.icon.size[size],
+    fontSize: size === 'inherit' ? size : theme.typography.icon?.size[size],
```
Comment (danidewitt) [inline packages/recipe/src/components/EzIcon/EzIcon.tsx:23]: 👍

File: packages/recipe/src/components/EzChip/Implementations/EzChipMui.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ const EzChipMui = forwardRef<Ref, EzChipProps & {children?: null}>(
     );
 
     const DeleteIcon = (
-      <div style={{display: 'flex', fontSize: '12px'}}>
+      <div style={{display: 'flex', fontSize: '12px'}} data-testid="DeleteIcon">
```
Comment (danidewitt) [inline packages/recipe/src/components/EzChip/Implementations/EzChipMui.tsx:35]: 👍

**Review by danidewitt — APPROVED** [review]

File: packages/recipe/.storybook/preview-head.html [hunk 3]
```diff
@@ -1 +1,56 @@
...
+  rel="stylesheet"
+/>
+<link
+  href="https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap"
+  rel="stylesheet"
+/>
+
+<style>
+  body {
+    font-family: Montserrat, sans-serif;
```
[the remark this answers is not in the document]
Comment (noranda) [reply packages/recipe/.storybook/preview-head.html:12]: We actually use both in our current doc site so I think I was trying to make them match. We can make the fonts the same though if we want!

File: packages/recipe/.storybook/theme.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+const baseTheme = {
+  // Brand
+  brandTitle: 'Recipe',
+  brandUrl: 'https://ezcater.github.io/recipe/',
+  brandImage: 'https://i.imgur.com/uIjnFxK.png',
+  brandTarget: '_self',
+
+  // Typography
+  fontBase: 'Lato, "Helvetica Neue", Arial, Helvetica, sans-serif',
```
[the remark this answers is not in the document]
Comment (noranda) [reply packages/recipe/.storybook/theme.js:12]: See above.

File: packages/recipe/docs/Welcome.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import {Meta, Story} from '@storybook/addon-docs';
+import Intro from './components/Intro/Intro';
+import metaParameters from './utils/metaParameters';
+
```
[the remark this answers is not in the document]
Comment (noranda) [reply packages/recipe/docs/Welcome.stories.mdx:4]: Yeah I noticed the same and extracted `metaParameters`, which is as far as I could extract and still have `storybook` work correctly. I can look into this further in the next iteration.

**Review by pgrippi — APPROVED** [review]

**Pull request merged**